services/search/pkg/search/events.go

```diff
@@ -0,0 +1,185 @@
+package search
+
+import (
+	"context"
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	user "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	rpcv1beta1 "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/v2/pkg/events"
+	"github.com/cs3org/reva/v2/pkg/storagespace"
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	"github.com/owncloud/ocis/v2/services/search/pkg/config"
+	"github.com/owncloud/ocis/v2/services/search/pkg/content"
+	"github.com/owncloud/ocis/v2/services/search/pkg/engine"
+)
+
+type eventHandler struct {
+	logger    log.Logger
+	engine    engine.Engine
+	gateway   gateway.GatewayAPIClient
+	extractor content.Extractor
+	secret    string
+}
+
+func HandleEvents(eng engine.Engine, extractor content.Extractor, gw gateway.GatewayAPIClient, bus events.Stream, logger log.Logger, cfg *config.Config) error {
+	ch, err := events.Consume(
+		bus,
+		"search",
+		events.ItemTrashed{},
+		events.ItemRestored{},
+		events.ItemMoved{},
+		events.ContainerCreated{},
+		events.FileTouched{},
+		events.FileVersionRestored{},
+		events.FileUploaded{},
+		events.UploadReady{},
+	)
+	if err != nil {
+		return err
+	}
+
+	go func(eh *eventHandler, ch <-chan interface{}) {
+		for e := range ch {
+			switch ev := e.(type) {
+			case events.ItemTrashed:
+				eh.logger.Debug().Interface("event", e).Msg("marking document as deleted")
+				eh.trashItem(ev.ID)
+			case events.ItemMoved:
+				eh.logger.Debug().Interface("event", e).Msg("resource has been moved, updating the document")
+				eh.moveItem(ev.Ref, &user.User{Id: ev.Executant})
+			case events.ItemRestored:
+				eh.logger.Debug().Interface("event", e).Msg("marking document as restored")
+				eh.restoreItem(ev.Ref, &user.User{Id: ev.Executant})
+			case events.ContainerCreated:
+				eh.logger.Debug().Interface("event", e).Msg("resource container created, updating the document")
+				eh.upsertItem(ev.Ref, &user.User{Id: ev.Executant})
+			case events.FileTouched:
+				eh.logger.Debug().Interface("event", e).Msg("resource has been changed, updating the document")
+				eh.upsertItem(ev.Ref, &user.User{Id: ev.Executant})
+			case events.FileVersionRestored:
+				eh.logger.Debug().Interface("event", e).Msg("resource version restored, updating the document")
+				eh.upsertItem(ev.Ref, &user.User{Id: ev.Executant})
+			case events.FileUploaded:
+				if cfg.Events.AsyncUploads {
+					return
+				}
+
+				eh.logger.Debug().Interface("event", e).Msg("resource upload ready, updating the document")
+				eh.upsertItem(ev.Ref, &user.User{Id: ev.Executant})
+			case events.UploadReady:
+				eh.logger.Debug().Interface("event", e).Msg("async resource upload ready, updating the document")
+				eh.upsertItem(ev.FileRef, ev.ExecutingUser)
```

Comment:
Please remove all the debug log statements and add instead one
```
eh.logger.Debug().Interface("event", e).Msg("updating index")
```
The type of the event is clear because you log it.